In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv('../../Data/raws/WA_Fn-UseC_-Telco-Customer-Churn.csv')

In [11]:
print(df['Partner'].unique()) 
print("\n")
print(df['Dependents'].unique())  

['Yes' 'No']


['No' 'Yes']


In [19]:
((df['Dependents'] == 'Yes') & (df['Partner'] == 'No')).sum()

np.int64(361)

In [20]:
(
    (df['Dependents'] == 'Yes') &
    (df['Partner'] == 'No')
).sum() / (df['Dependents'] == 'Yes').sum()

np.float64(0.17109004739336492)

ต้องระวัง! Partner กับ Dependent มีความไม่สัมพันทธ์กัน 361 อาจจะ เป็นกลุ่มคนที่ (พ่อ/แม่เลี้ยงเดี่ยว, คนดูแลพ่อแม่, หย่าร้างแต่ยังมีลูก, อื่น ๆ)

In [29]:
df["tenure"].unique()

array([ 1, 34,  2, 45,  8, 22, 10, 28, 62, 13, 16, 58, 49, 25, 69, 52, 71,
       21, 12, 30, 47, 72, 17, 27,  5, 46, 11, 70, 63, 43, 15, 60, 18, 66,
        9,  3, 31, 50, 64, 56,  7, 42, 35, 48, 29, 65, 38, 68, 32, 55, 37,
       36, 41,  6,  4, 33, 67, 23, 57, 61, 14, 20, 53, 40, 59, 24, 44, 19,
       54, 51, 26,  0, 39])

In [30]:
df["PaymentMethod"].unique()    

array(['Electronic check', 'Mailed check', 'Bank transfer (automatic)',
       'Credit card (automatic)'], dtype=object)

Electronic check = จ่ายด้วยเช็คอิเล็กทรอนิกส์ (ตัดเงินออนไลน์ แต่ไม่ auto)

Mailed check =	ส่งเช็คทางไปรษณีย์

Bank transfer(automatic) =	โอนผ่านธนาคารแบบหักเงินอัตโนมัติ

Credit card (automatic) = ตัดบัตรเครดิตอัตโนมัติ

In [34]:
print(df["MonthlyCharges"].unique())

print("\n")

print("Min MontlyCharge:",df["MonthlyCharges"].min())
print("Max MontlyCharge:",df["MonthlyCharges"].max())

[29.85 56.95 53.85 ... 63.1  44.2  78.7 ]


Min MontlyCharge: 18.25
Max MontlyCharge: 118.75


In [41]:
print("Data Type of TotalCharges:  ",df["TotalCharges"].dtype)
df["TotalCharges"] = pd.to_numeric(df["TotalCharges"], errors="coerce")

print(df["TotalCharges"].unique())

print("\n")

print("Min TotalCharges:",df["TotalCharges"].min())
print("Max TotalCharges:",df["TotalCharges"].max())

Data Type of TotalCharges:   float64
[  29.85 1889.5   108.15 ...  346.45  306.6  6844.5 ]


Min TotalCharges: 18.8
Max TotalCharges: 8684.8
